src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java

```diff
@@ -182,18 +183,25 @@ void postResultsToWorkspace(String results, String workspace, int buildNumber)
   @VisibleForTesting
   void validateAndFailBuild(String results, TaskListener listener) throws IOException {
     listener.getLogger().println(results);
-  LOGGER.log(Level.INFO, "Results::" + results);
+    LOGGER.log(Level.INFO, "Results::" + results);
     if (isFailBuild()) {
-      JSONObject jsonObject = new JSONObject(results);
-      JSONArray violations = jsonObject.optJSONArray("failed_policies");
-      if (violations != null && violations.length() > 0) {
-        for (int i = 0; i < violations.length(); i++) {
-          JSONObject violation = violations.optJSONObject(i);
-          String severity = violation.optString("severity");
-          if (severity != null && ("HIGH".equals(severity.toUpperCase(Locale.ROOT)) || "CRITICAL".equals(severity.toUpperCase(Locale.ROOT)))) {
-            throw new AbortException("Zscaler IaC scan found violations, they need to be fixed");
+      try {
+        JSONObject jsonObject = new JSONObject(results);
+        if(jsonObject.keySet().contains("no_resources_message")) {
+          return;
+        }
+        JSONArray violations = jsonObject.optJSONArray("failed_policies");
+        if (violations != null && violations.length() > 0) {
+          for (int i = 0; i < violations.length(); i++) {
+            JSONObject violation = violations.optJSONObject(i);
+            String severity = violation.optString("severity");
+            if (severity != null && ("HIGH".equals(severity.toUpperCase(Locale.ROOT)) || "CRITICAL".equals(severity.toUpperCase(Locale.ROOT)))) {
```

Comment:
We should consider any severity as failure.

Reply:
Sure @jragula-zs